Calculator/Calculator/CalculatorItemQueue.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  CalculatorItemQueue.swift
+//  Calculator
+//
+//  Created by Mangdi on 2022/09/20.
+//
+
+import Foundation
+
+struct CalculatorItemQueue<T>: CalculateItem {
+    var front: Node<T>?
+    var rear: Node<T>?
+    
+    func isCheckEmpty() -> Bool {
+        if front == nil {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    mutating func enQueue(_ item: T) {
+        let newNode: Node<T>? = Node(data: item)
+        
+        if isCheckEmpty() {
+            front = newNode
+            rear = newNode
+        } else {
+            rear?.link = newNode
+            rear = newNode
+        } 
+    }
+    
+    mutating func deQueue() -> T? {
+        if !isCheckEmpty() {
+            guard let data = front?.data else { return nil }
+            front = front?.link
+            changeRear()
+            
+            return data
+        } else {
+            print("비어있습니다")
+            return nil
+        }
+    }
+    
+    mutating func changeRear() {
+        if isCheckEmpty() {
+            rear = nil
+        }
+    }
+    
+    func peek() -> T? {
+        if !isCheckEmpty() {
+            let data = front?.data
+            return data
+        } else {
+            return nil
+        }
```

Comment:
아래와 같이 구현 가능할 것 같네요~!?
참고 부탁드려요!
```suggestion
        return front?.data
```

Reply:
헐 맙소사 이게 이렇게 줄어질수있었군요..!
peek()메서드가 nil을 반환할수도있기때문에 이런식으로 짤수있네요..
한 수 배워갑니다!